### DISCUSSION TITLE
Computed properties attrtibutes partially broken?

### DISCUSSION THREAD

**Discussion by stychos** [livewire/livewire#8492] [1. Help] [answered]
Hello, maybe someone can approve or disapprove my observation here? I have a full-page Livewire component which has few other children components. Full-page component has Modelable binding with the one of its childrens (a filter) and pass this model to other childrens, which are using it as Reactive properties. Code for the parent page:

```php
use App\Dtos\Filters\User as Filter;
use Livewire\Component;

class App extends Component
{
	#[Session]
	public Filter $filter;

	public function mount()
	{
		$this->filter ??= new Filter(
			user: $this->user->id,
		);
	}
}
```

Code for the children component:
```php
use App\Dtos\Filters\User as Filter;
use Livewire\Component;

class Directory extends Component
{
	#[Reactive]
	public Filter $filter;

	#[Computed(persist: true)]
	public function user(): User
	{
		return $this->filter->user() ?? auth()->user();
	}
	
	// ...
}
```

Filter DTO is a simple Wireable without any complex logic:
```php
namespace App\Dtos\Filters;

use App\Models\User as Model;
use Livewire\Wireable;

class User implements Wireable
{
	public function __construct(
		public ?int $user = null,
		// ...
	) {
	}

	public function toLivewire()
	{
		return [
			'user' => $this->user,
			// ...
		];
	}

	public static function fromLivewire($values)
	{
		return new static(
			user: $values['user'],
			// ...
		);
	}

	public function user(): ?Model
	{
		return Model::find($this->user);
	}
}
```

Everything is going smooth and perfect until I trigger any event on the children component, the I am getting `Livewire\Exceptions\PropertyNotFoundException: Property [$user] not found on component: [pages.agent.directory]`. This happens even with `persist: true` attribute's option.

If I change attribute-based syntax to the legacy `public function getUserAttribute()`, then everything works fine.
Is it some bug / flaw / documentation lack in Livewire or I am doing something wrong? Thanks.

**Comment by joshhanley** [comment]
@stychos Make sure you import the computed attribute class/namespace otherwise it won't work

**Reply by stychos** [reply]
For sure it's imported, it works on the page load.

**Reply by joshhanley** [reply, marked as answer]
@stychos Ah right, wasn't in your code you shared. Can you share your full component classes and views?

**Reply by stychos** [reply]
> @stychos Ah right, wasn't in your code you shared. Can you share your full component classes and views?

Quite strange, but now I prepared example clean project with such structure and everything works fine in it. So, looks like the problem is on my side. Thanks a lot for your help!
